src/Particular.Analyzers/Cancellation/CancellationTryCatchAnalyzer.cs

```diff
@@ -0,0 +1,244 @@
+                .Where(node => node.ToString() == expression);
+
+            return tokens.Any();
+        }
+
+        static bool HasFilterIncludingCaughtException(CatchClauseSyntax catchClause)
+        {
+            var identifier = catchClause.Declaration?.Identifier.Text;
+            if (identifier == null)
+            {
+                // just `catch (Exception)` or even `catch { }` - no variable name
+                return false;
+            }
+
+            if (catchClause.Filter == null)
+            {
+                return false;
+            }
+
+            var filterIdentifiers = catchClause.Filter.DescendantNodes()
+                .OfType<IdentifierNameSyntax>()
+                .Where(nameSyntax => nameSyntax.Identifier.ValueText == identifier);
+
+            return filterIdentifiers.Any();
+        }
+
+        static string GetFirstCancellationTokenExpressionOrDefault(SyntaxNodeAnalysisContext context, TryStatementSyntax tryStatement)
+        {
+            // Because we are examining all descendants, this may result in false positives.
+            // For example, a nested try block may contain cancellable invocations and
+            // a related catch block may swallow OperationCanceledException.
+            // Or, an anonymous delegate may contain cancellable invocations but
+            // may not actually be executed in the try block.
+            // However, these are edge cases and would be complicated to analyze.
+            // In these cases, either the fix can be redundantly applied, or the analyzer can be suppressed.
+            var tryBlockCalls = tryStatement.Block.DescendantNodes().OfType<InvocationExpressionSyntax>();
+
+            var distinctTokenExpressions = GetCancellationTokenExpressions(tryBlockCalls, context)
+                .Distinct()
+                .ToImmutableArray();
+
+            if (distinctTokenExpressions.Length > 1)
+            {
+                context.ReportDiagnostic(DiagnosticDescriptors.MultipleCancellationTokensInATry, tryStatement.TryKeyword);
+            }
+
+            return distinctTokenExpressions.FirstOrDefault();
+        }
+
+        static IEnumerable<string> GetCancellationTokenExpressions(IEnumerable<InvocationExpressionSyntax> calls, SyntaxNodeAnalysisContext context)
+        {
+            foreach (var call in calls)
+            {
+                if (call.Expression is MemberAccessExpressionSyntax memberAccess && memberAccess.IsKind(SyntaxKind.SimpleMemberAccessExpression))
+                {
+                    if (memberAccess.Name.Identifier.ValueText == "ThrowIfCancellationRequested")
+                    {
+                        if (context.SemanticModel.GetTypeInfo(memberAccess.Expression, context.CancellationToken).Type.IsCancellationToken())
+                        {
+                            yield return memberAccess.Expression.ToString();
+                            continue;
```

Comment:
I added this as a performance tweak, since there's no need to analyse this call further.